### DISCUSSION TITLE
【第4回】2024年03月31日

### DISCUSSION THREAD

**Discussion by kagomen** [first-contributions-ja/first-contributions-ja.github.io#136] [【メンテナー必読】週報]
## 目次
例）
- [トピック: これは例です]
  - 下記のスレッド内にて決定→：[Aにする]
- [トピック: これは例です]
  - [B]に関して、再度MTGで共有・確認する！

---

- [トピック: Tailwind CSSのサイズの使用についての確認](https://github.com/first-contributions-ja/first-contributions-ja.github.io/discussions/136#discussioncomment-8948525)
  - 基本余白は、16の倍数でいく！
  - [px]は使わない方向にするので、後にリファクタする！
  - 現在のサイトにある[Step]は、コンポーネント切って、ループする！
- [トピック: CONTRIBUTING.mdに追記したい内容について](https://github.com/first-contributions-ja/first-contributions-ja.github.io/discussions/136#discussioncomment-8949139)
  - 新しく決まったことがあればWikiに書き加えていく
  - バージョンリリースなど大きな区切りで、WikiからCONTRIBUTING.mdに転記すべきものがあるかを確認->転記する
- [トピック: developブランチについて](https://github.com/first-contributions-ja/first-contributions-ja.github.io/discussions/136#discussioncomment-8949400)
  - 一旦、今まで通りmain + 作業用ブランチの方式で！
- [トピック: バージョン2のデザイン担当について](https://github.com/first-contributions-ja/first-contributions-ja.github.io/discussions/136#discussioncomment-8949812)
  - 簡易的なデザインカンプをみんなで作って持ち寄り、みんなの案をもとに1人がデザインをまとめ、デザインカンプを完成させる
- [トピック: デザインはスマホ版から考えるべきか？PC版から考えるべきか？](https://github.com/first-contributions-ja/first-contributions-ja.github.io/discussions/136#discussioncomment-8951186)
  - 下記のスレッド内にて決定→：デザインは先に用意する！実装自体はスマホから組んでいく！
- #145 
  - 採用
  - 事前にMTGで話したいことはDiscussionsに投稿、意見を出し合う
  - MTGで話したいトピックを目次に追加、結論はトピック追加した人がまとめる
  - 決定したことはさらっと確認、必要なものは話して詰めていく

---

## 次の1週間でやること

今後の大まかな流れは下記：
- バージョン１リリースまで：
  - ドッグフーディングしてみる
  - ドッグフーディングにて、見つかった修正箇所などをリストアップする（discussionの専用スレ → issue化）
  - 上記の修正＋スクショの追加
  - バージョンのタグと、リリースノートを作る！
- バージョン1.0.n：リファクタリングなどの修正で、バージョン２の開発はまだ行わない
- バージョン２の仕様決め：
  1. コンセプト決め：リポジトリのコンセプトやストーリー性、Contributors.jsonに入力してもらうデータを決める
  2. デザインコンペ：それを元に、個別ページ、トップの表示（カード？）などのデザインをみんなでそれぞれ作ってみる
  3. デザインカンプ用意：それぞれのデザインの良いところを取ったり混ぜたりして、1人がベースを整えてまとめる
  4. バージョン２の実装！

**その他**：
- issuesは、以前の優先度の高いタスクのまとめが、ピン留めしてある方式の方がやりやすいかも？なので、戻してみてもいいかも！

---

以下フォーマット（トピックを追加したら、上記の目次にも追加してください！）
```
## トピック: タイトル
- 概要

## 結論: 一言でまとめる
- 結論

## 関連 Issues、Discussions
```

**Comment by kagomen** [comment]
## トピック: Tailwind CSSのサイズの使用についての確認
  - 通常のCSSだと8の倍数（細かく調整したい場合は4の倍数）を基準に`px`を指定していくが、Tailwindを用いた場合はどうするのか？を確認したい
    - 個人的には、既にTailwind側がそのような数値になるよう調整してくれているので、好きな数値を使っていいと思っていた！
    - また、大きな数値は`m-64`などとするより`[--px]`の方がわかりやすいと思っているが、そのあたりの認識のすり合わせもしたい！

## 結論: 一言でまとめる
- 基本余白は、16の倍数でいく！
- [px]は使わない方向で！
  - なので、リファクタする！
- 現在のサイトにある[Step]は、コンポーネント切って、ループする！

## 関連 Issues、Discussions
#131 
https://github.com/first-contributions-ja/first-contributions-ja.github.io/pull/131#discussion_r1541554798

**Reply by pss-aileen** [reply]
そうですね、すり合わせが必要ですね...！
MTGの前に一応自分の考え方をシェアしておきます。

### まず8の倍数、16の倍数について → 16の倍数基準が良いと思う
- 8の倍数が良いとはよく言われますが、8はすごく量が多くて悩みの原因になります
  - 8の場合: 8, 16, 24, 32, 40, 48, 56, 64, 72, 80, 88, 96...
  - 16の場合: 16, 32, 48, 64, 80, 96
- そのため、基本16の倍数で余白をあけるように考えると、余白の取り方が綺麗になる
- ただし、文字の並びでは、16pxでは余白が大きくなりすぎる
  - その場合は、8の倍数を使っても良い（STEP1と本文の隙間とか、ボタンのアイコンと文字の余白とか）
  - 16px、8pxで考えてもなお余白が広すぎるなと考えた時は、4の倍数、2の倍数と小さくしていく。
- **自分の意見まとめ: 基本的な余白は16px で考えて、細かい文字の隙間は16px→8xp→4px→2pxでしていくと良い**
  - 文字の大きさも同じように考えています（こちらはあまり16に固執はしていませんが、↑を基準にしています）

### ある程度の余白のすり合わせができればCSSを何か別で設定して良いかも
- TailwindCSSまだ調べ切ってませんが、普段使う余白が決まれば `.layout-space-1(16px)`, `.layout-space-2(24px)` などとしても良いかもしれません
- pxにすると、結局あとで万が一変えたい場合全てを変更する必要があるので（今のtailwindでも同じですが...）、確定しているものは、CSSにするか、どこかで定義して、それを使うのが良いかと思います

### 参考
8の倍数とは書いているが、基本16の倍数が多い
https://yuyakinoshita.com/blog/2019/02/10/design-by-multiple-of-8/

![スクリーンショット 2024-03-29 16 11 08](https://github.com/first-contributions-ja/first-contributions-ja.github.io/assets/140990100/f7346771-3873-45a2-abfa-59cc830104ba)

![スクリーンショット 2024-03-29 16 11 24](https://github.com/first-contributions-ja/first-contributions-ja.github.io/assets/140990100/e521ed5c-74cf-4ab1-a842-e1eabc280dbc)

**Reply by kagomen** [reply]
共有ありがとうございます！
大きなレイアウトの余白を考える場合は16の倍数が良さそうです！🙏

> ある程度の余白のすり合わせができればCSSを何か別で設定して良いかも

こちらについても話したいですね。
自分は、Tailwindの数値に関しては非常に面倒だなーと感じます。個人の時は`m-[32px]`のように使用してます・・・

> 16px→8xp→4px→2pxでしていくと良い

この前提さえ共有していれば、無駄な数値の対応表を覚えなくて済むのに・・・
個人的な理想を言えば、`m-32`で`margin: 32px`が設定されることです

**Reply by pss-aileen** [reply]
こちらですが、CSSの設定（mt-32とわかりやすくすること）はデザインが出来上がった後に詳細決めたほうがいいかもです。

デザインが決まればよく使うマージンがハッキリするので、
それが決まってから `mt-32` とか、そういうクラスを設けて視覚的にもわかりやすいものを作れればと思います！
そもそもデザインが決まれば mt-32 などは作らなくてよいかもしれません...！

あ、でもTailwindCSSの記述がわかりにくい、ということですよね。
`mt-2` といいならが `8px` だったり、`mt-3` が `12` だったり...ということですよね？

個人的にはこのTailwindCSSのルールはある程度慣れたので、mt-2（8px）となるのはあまり気になりません😭

**Reply by kagomen** [reply]
> TailwindCSSの記述がわかりにくい、ということですよね。

ですです！

> 個人的な理想を言えば、m-32でmargin: 32pxが設定されることです

こういう設定をしたいというわけではないです！すみません🙏💦

慣れるんですね……。
なんで既にわかっている数字を4で掛けて計算し直さないといけないんだろう、と。かなり嫌いポイントです。😠

**Reply by kazzyfrog** [reply]
個人的には、[--px]は使わないほうがいいかも、と思ってます！
あえてtailwindの制限（柔軟にpxを指定できない）に従う方が、tailwindのメリットを享受できるのかな、と思っています！

めっちゃ暇な時に[この記事](ある程度の余白のすり合わせができればCSSを何か別で設定して良いかも)読んで見てもいいかもです！
長いですが、少しtailwindに詳しくなれます！

上記の記事からの引用
> いつぞやのバージョンアップで、w-[57px]のように、[ ]構文を用いることで制限に縛られない自由な値を使える機能が追加されました。これは明らかに大きな抜け穴であり、これまで説明してきたTailwindの意義に真っ向から反しています。筆者もこれを知ったとき、Tailwindがどのような方向性を目指しているのかよく分からなくなりました。
Tailwindをデザインシステムのために使っている限り、この機能は明らかに濫用すべきではありません。TypeScriptで言うところのanyやasに相当するものです。

**Reply by kazzyfrog** [reply]
それと、個人的には、tailwindの思想というか、ベストプラクティスにならって、
細かくコンポーネントを分割する方向で、UIを整理していけたらいいなって思ってます！
（これは個人の好みが強いです）

例えば、`<Container>`コンポーネント（セクションの余白などを統一する用）を作成し、サイトのセクションをラップすることで、全部の余白など統一できます！
[こちらのリポジトリ](https://github.com/web3templates/nextly-template/blob/main/components/container.js)の実装を引用すると、以下のようなイメージ！
Container コンポーネント：
```jsx
const Container = (props) => {
  return (
    <div
      className={`container p-8 mx-auto xl:px-0 ${
        props.className ? props.className : ""
      }`}>
      {props.children}
    </div>
  );
}

export default Container;
```
Container コンポーネントでラップして作成するコンポーネントの例：
```jsx
const Cta = () => {
  return (
    <Container>
      <div className="flex flex-wrap items-center justify-between w-full max-w-4xl gap-5 mx-auto text-white bg-indigo-600 px-7 py-7 lg:px-12 lg:py-12 lg:flex-nowrap rounded-xl">
        <div className="flex-grow text-center lg:text-left">
          <h2 className="text-2xl font-medium lg:text-3xl">
            Ready to try-out this template?
          </h2>
          <p className="mt-2 font-medium text-white text-opacity-90 lg:text-xl">
            Don&apos;t let your visitors see a poor landing.
          </p>
        </div>
        <div className="flex-shrink-0 w-full text-center lg:w-auto">
          <a
            href="https://github.com/web3templates"
            target="_blank"
            rel="noopener"
            className="inline-block py-3 mx-auto text-lg font-medium text-center text-indigo-600 bg-white rounded-md px-7 lg:px-10 lg:py-5 ">
            Download for Free
          </a>
        </div>
      </div>
    </Container>
  );
}

export default Cta;
```

**Reply by kazzyfrog** [reply]
同様に、H2やH3などは、`<Heading>`コンポーネントを作れば、
どこからでも呼び出せて、上下の余白なども１箇所で管理できます！

[こちらのリポジトリ](https://github.com/muxinc/video-course-starter-kit/blob/main/components/Heading.tsx)より引用

```jsx
type Props = {
  children: React.ReactNode;
  as?: 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6';
};

const Heading = ({ children, as = 'h1' }: Props) => {
  switch (as) {
    case 'h1':
      return <h1 className='text-4xl font-bold mb-8 text-slate-700 font-cal'>{children}</h1>;
    case 'h2':
      return <h2 className='text-3xl font-bold mb-4 text-slate-600 font-cal'>{children}</h2>;
    case 'h3':
      return <h3 className='text-2xl font-bold mb-4 text-slate-600 font-cal'>{children}</h3>;
    case 'h4':
      return <h4 className='text-xl font-bold mb-4 text-slate-600 font-cal'>{children}</h4>;
    case 'h5':
      return <h5 className='text-lg font-bold mb-4 text-slate-600 font-cal'>{children}</h5>;
    case 'h6':
      return <h6 className='text-base font-bold mb-4 text-slate-600 font-cal'>{children}</h6>;
  }
};

export default Heading;
```

こういうコンポーネントの分割で、細かく管理するのもいいかもです！
クラス長くなっても、コンポーネント内のコード量自体が減れば、そこまで見づらくならないかも？と少し思ってます！🙌

**Reply by kagomen** [reply]
こちら https://blog.uhy.ooo/entry/2022-10-01/tailwind/ 読んでみました。とても興味深かったです！ありがとうございます🙏✨
読んでて思ったのが、フレームワークって思想なんだなーってことでした。
自分の中では最初にデザインという目的があって、それを実現する手段としてCSSが存在する、という認識があり、Tailwindもそれと同じように考えていたのですが、ちょっと違うのですね。
Tailwindはプロジェクト（プログラミングのしやすさともちょっと違う）を第一の目的にしていて、デザインはその次なんだなと感じました。それがユーティリティファーストということなのかな、と思ったりもしました。（憶測ですが💧）

**Reply by kazzyfrog** [reply]
ReactとTailwindと、デザイン？の勉強にもなるので、色々コンポーネントの分割とか、CSS変数使ったりとか、
模索しながら、しっくりくる構成を見つけてもいいかもですね！👍

**Comment by kagomen** [comment]
## トピック: CONTRIBUTING.mdに追記したい内容について
- [CONTRIBUTING.md](https://github.com/first-contributions-ja/first-contributions-ja.github.io/blob/main/docs/CONTRIBUTING.md)に、プルリク承認の方法も記載したい
  - 記載する場合、行程を再度確認したい
  - 以前のMTGでSquashするときにタイトルをどうするか、などを聞いた気がするのですが、忘れてしまったため、文章として残しておきたい・・・

- 同時に、バージョン1で区切りがついた段階で、https://github.com/first-contributions-ja/first-contributions-ja.github.io/discussions/52 で集まったコーディング規則などもCONTRIBUTING.mdに転記する

## 結論: 
- 新しく決まったことがあればWikiに書き加えていく
- バージョンリリースなど大きな区切りで、必要があればCONTRIBUTING.mdに転記する

## 関連 Issues、Discussions

**Reply by pss-aileen** [reply]
squash時は、コミットの内容の代表的なもの？をコミットメッセージ風に書くで良かったと思います。
なので、レスポンシブの件だと「🎨 style:レスポンシブデザインの実装」そのままでOKかと！

マニュアル的なものはWikiを使ってみても良いかなと考え始めました
Wikiにコーディング規約まとめていって、
バージョンリリースしたり、大きな節目ごとにCONTRIBUTING.mdなどに追記でも良いかも。
でも、追記してしまうとCONTRIBUTINGが膨大にもなりそうなのでWiki使ってみたいです！

**Reply by kagomen** [reply]
> squash時は、コミットの内容の代表的なもの？をコミットメッセージ風に書くで良かったと思います。

ありがとうございます🙏了解です！
ちなみにタイトル下のテキスト欄は、自動記入で複数のコミットメッセージが羅列されますが、こちらはプルリクエストの文章に変更する感じでしたっけ・・・？🌀

> マニュアル的なものはWikiを使ってみても良いかなと考え始めました

Wiki良いですね👍使ってみたいです！
大きな流れはCONTRIBUTING.md、コーディング規約はWikiって感じで良いかもですね！

**Reply by pss-aileen** [reply]
> ちなみにタイトル下のテキスト欄は、自動記入で複数のコミットメッセージが羅列されますが、こちらはプルリクエストの文章に変更する感じでしたっけ・・・？🌀

テキスト欄は、自動記入のままで良いかと思います！（ここ話し合ったか記憶喪失です）
コミットメッセージの横の「・・・」で詳細を確認して、「あ、これね、これはここね」と思い出せるかと思うので、そのままで良いと思います！

**Reply by kagomen** [reply]
了解です！🫡
ありがとうございます！！！✨

**Reply by kagomen** [reply]
結論をまとめようとしたら不明点が出てきたので質問です！

結局、
バージョン●●のスレッドにどんどん書いていって、集まったらCONTRIBUTING.mdとWikiに転記、って感じで良いんですかね❓
それとも、コーディング規約などはWikiに書いていった方が良いですか？
Wikiは更新されても通知が来ず（私だけ？）目に止まりにくいので（イシューやディスカッションでさえ見逃しがある。。。）、多少面倒でも前者の方が良いかな、と思ったのですが、どうでしょうか？👀

**Reply by kazzyfrog** [reply]
Wikiいいですね！

こういうのはどうですか？
- 閲覧：Wikiに、現時点で決まっている規約が書いてあるので、ここを見ればOK！
- 提案：おそらく、discussionやPRでのやり取り内で、相談・提案して、こういうふうにしていこう！というのが決まる！
- 決まったらWikiに書き加える！

そして、CONTRIBUTING.mdには、
代表的な注意書き（コミットメッセージやPRのテンプレURLとか、もしくはおおまかな方針など？、、）を書いて、
「それ以外はWikiを参考してください！まだ決まりがなければ、discussionやPRの際に、相談・提案してください！」と記載する、とか？

**Reply by kazzyfrog** [reply]
マージに関するまとめ：

基本は[Wiki](https://github.com/first-contributions-ja/first-contributions-ja.github.io/wiki/プルリクエストからマージの流れ#プルリクエストのマージ方法)の通り👍

- コミットが1つの場合、Merge pull requestで、マージする際のメッセージはデフォルトのまま
- 複数のコミットをまとめたい場合、Squash Mergeで、タイトルをコミットメッセージに書き換える
  - コミットメッセージはスタイル通り：「 :books: docs: READMEの作成 #00 」

微妙なラインを、以下に記載しておきます！

**Reply by kazzyfrog** [reply]
**事例①：コミット複数だが、タスクも複数で、記録に残したいコミットの場合→Merge pull request**

#124 で、prettier-plugin-tailwindcssを導入しました！
これはコミットを２つ積んでます：
- prettier-plugin-tailwindcssの導入
- npm run format

このように、あえて分けたい場合は、Squash ではなく、Merge pull requestでもOKだと思ってます！

理由は、formatによる整形で、複数のファイルに変更が加えられています！
そのコミットメッセージが、`prettier-plugin-tailwindcssの導入`なのは、厳密には正しくではないからです！
適切に、記録したいまとまりを区切れているのであれば、OK！

**Reply by kazzyfrog** [reply]
**事例②：雑コミットが複数なら、Squash Merge**

#89 で、READMEを作成しました！
これは、コミットを７つ積んでます。
しかし、タスクが複数でもないし、適切に区切って、記録を残したいわけでもないです。。
セーブ的にコミット打って、共有するためにコミットして、PRを出して、フィードバックを元に再度修正のコミット積んで、、、
って感じでした！

この場合は、mainに複数の雑コミット含めるべきでないので、Squash MergeでOKだと思ってます！
（もしくはリベースして、2とか3とか、必要な分だけに厳選してもOK）

**Reply by pss-aileen** [reply]
> - 閲覧：Wikiに、現時点で決まっている規約が書いてあるので、ここを見ればOK！
> - 提案：おそらく、discussionやPRでのやり取り内で、相談・提案して、こういうふうにしていこう！というのが決まる！
> - 決まったらWikiに書き加える！

取り急ぎですが、こちらで良いと思います！

**Reply by kagomen** [reply]
細かい事例の説明、ありがとうございます🙏✨

- タスクが1つの場合:
  - コミットが1つの場合: Merge
  - コミットが複数の場合: Squash
- タスクが複数の場合: Merge

理由: タスク単位でコミットメッセージを統一したいから

ってことで合ってますでしょうか？
ちなみにタイトルの変更にMergeとSquashで差があるのはなぜなんでしょうか……🌀

**Reply by kazzyfrog** [reply]
> ってことで合ってますでしょうか？

その認識であっていると思います！
イシューの粒度も、細かく分割して、
基本1タスク+ 分ける方が、かえってややこしくなるような関連する細かいこと
とかにすると、簡潔になりそうですね！

> タイトルの変更にMergeとSquashで差があるのはなぜなんでしょうか……🌀

[コミットの履歴](https://github.com/first-contributions-ja/first-contributions-ja.github.io/network)を見てみると、
Squashの場合は、mainにそのままコミットを追加しているように見えます！
Merge pull requestの場合は、作業したコミットの追加に加えて、mainにマージしました！というコミット（マージコミット）も作成されます！

まとめ：
- Squashの場合は、まとめたコミットを、メッセージを編集してmainに追加する
- Merge pull requestの場合は、作業用ブランチでのコミットがそのまま反映され、Merge pull requestのメッセージは、マージしました！ということを表しているだけなので、デフォルトのままでOK！

**Reply by kagomen** [reply]
> Merge pull requestのメッセージは、マージしました！ということを表しているだけ

なるほどです！理解しました！
ありがとうございます🙏✨

**Comment by kagomen** [comment]
## トピック: developブランチについて
バージョン１の実装終了後、バージョン２の実装はmainブランチの他にdevelopブランチを作成する必要がある（はず？）
その場合のブランチの切り方について

## 結論: 一言でまとめる
- 一旦、今まで通りで！

## 関連 Issues、Discussions

**Reply by kazzyfrog** [reply]
これは、どちらでも大丈夫だと思います！

1. mainブランチと、作業用ブランチで今まで通り進めるパターンか、
2. mainブランチ（バージョン１リリース）と、developブランチ（バージョン２の開発）と、作業用ブランチで進めるパターン

１の場合は、今まで通りです！マージされたらすぐにサイトに反映されます！
２の場合は、developを今までのmainブランチのように扱い、作業用ブランチを切っていく！
そして、バージョン２のリリースのタイミングで、まとめてmainにマージする。
なので、サイトに反映されるのは遅くなります！

**Reply by kazzyfrog** [reply]
といいつつ、機能自体はそんなに多くないので、
話にあった通り、以下のような感じで良さそうですね：

- まずは、「バージョン1.0.n」で、リファクタリングなどの修正で、バージョン２の開発はまだ行わないので、今まで通りmainと作業用ブランチ
- そして、バージョン２の実装に関しては、developブランチを使って、バージョン２のリリースのタイミングで、まとめてmainにマージする！

もしくは、今まで通り、ずっとmainと作業用でもいいかもです！
UIとか早く確認できるので！自分はどっちでも！！👍

一応、バージョン２のタイミングで、今回と同じように、
どこかで区切っても、「バージョン２」のタグをつけることはできます！

**Reply by pss-aileen** [reply]
- ドックフーディングのドキュメント周りの修正までで1.0.0
- その他内部的なもの、細かいところは1.0.nでリファクタリング
- ↑をしながら、バージョン2の仕様を考えつつ...
- **developにて、バージョン2の開発、バージョン2リリースでmainにマージ**、かどうかは今から考える必要あり
- 参考
  - [バージョンにあれこれ考えを巡らせてみる #ポエム - Qiita](https://qiita.com/t_nakayama0714/items/c312bc5edcce6c214856#%E3%83%90%E3%83%BC%E3%82%B8%E3%83%A7%E3%83%8B%E3%83%B3%E3%82%B0%E3%83%AB%E3%83%BC%E3%83%AB)
  - [非エンジニアも知っておきたいソフトウェアのバージョン表記の書き方を解説 | ワードプレステーマTCD](https://tcd-theme.com/2022/09/how-to-versioning.html)

**Reply by kagomen** [reply]
ローンチするのがバージョン２なので、バージョン１〜バージョン２は今まで通りの実装方法で問題はないかと思うのですが、もしバージョン３があるのだとしたらdevelopブランチを作ることになると思います
その予行練習としてバージョン１〜２でもdevelopブランチを用いた開発を体験しておきたいかも？とは思いました！
が、いつもながら、どちらでも大丈夫です！

**Comment by kagomen** [comment]
## トピック: バージョン2のデザイン担当について
みなさんお時間があれば、デザインに興味ある人全員でデザインカンプ作る、というのはどうでしょう？👀という提案です。
これは個人的な話ですが、Figmaを使えるようになりたいです。
個人開発のときにやろうと思っても、結局ノートに雑に書くとかで終わってしまうので、強制力が欲しいです……。
そこで、三人寄れば文殊の知恵！みなさんが良ければ持ち寄ったデザインの中で良いとこどりをするのはどうでしょうか。
微妙ですかね……❓ご意見をうかがいたいです！

## 結論: 一言でまとめる
- 簡易的なデザインカンプをみんなで作って持ち寄る
- みんなの案をもとに1人がデザインをまとめ、デザインカンプを完成させる

## 関連 Issues、Discussions

**Reply by pss-aileen** [reply]
良いと思います！！！⭕️
良いとこどりも良いですが、ベース作る方はいた方がいいと思うので
ベースを @kagomen さんが作って、
みんなで相談しつつだったり、部分的な案などがあればFigmaで作って共有...的な感じでもよいかと思います！

**Reply by kagomen** [reply]
> 良いとこどりも良いですが、ベース作る方はいた方がいいと思うので

なるほどです……！
しかしその場合、デザイン全く自信ないので、ガラッと変更される前提でお願いしたいです……💧
↑なので作業全体がかなり遅れる可能性ありです……。

話変わりますが、たしかFigmaでデータの共有とか同時編集？とかもできますよね？👀
GitHub同様、そのようなFigmaの使い方にも慣れたいところです！

**Reply by pss-aileen** [reply]
あ、でもそうか

すみません、
良いところを集約でも良いのか...と思いました😂😂

ベース作ってと言いましたがカードや個別ページは色々案があるのでベースというより案を形にするところからですよね...

😂😂

早とちりしてました、すみません🙇

最終的な完成系を作ると考えたら
誰かベースがいた方が良いな〜という考えが先行してました

なのでみんな(？笑)で色々作って良いところ取ったり混ぜたりして、最後整えて...的な！

Figmaについては、招待とかすればコメントとか編集とかできたはずです！

**Reply by kagomen** [reply]
うーん、でも、通常であれば案出しって言葉を交わして行いますよね……。
それを各々わざわざデザイン作って持ってくるっていうのは、やっぱコスト大きいですかね……❓
Aileenさんもデザインお好きですし、Kazzyfrogさんもデザインしたい的なことおっしゃってましたし、自分もFigmaいじりたいというのがあり、コスト感については大丈夫かなーと思っていたのですが……

ちょっとそこらへんの意見もMTGで聞きたいです……！

**Reply by pss-aileen** [reply]
案出しは必ずしも口頭？でなくても良いと思います。（案出しがはっきり何をどうするかにもよるかと思いますが...！）
デザインであればなおさら作った方が良いかな...と思います。

完成系でなくていいのでワイヤーフレーム的なものでもいいし
図形とテキストを配置するだけでもいいし、
そういって目に見えるものとして作成して
それを事前に共有して、そこからMTGでつめていったりする方向が良い気がします。

確かにデザインを作るのはある程度時間を必要とするので大変かと思いますが...MTGで相談ですね...！

**Reply by kagomen** [reply]
> 完成系でなくていいのでワイヤーフレーム的なものでもいいし
> 図形とテキストを配置するだけでもいいし

たしかに！これなら気軽に図面でアイデア出しできそうです！
ありがとうございます！ぜひ、そうしたいです🙏✨

**Reply by kazzyfrog** [reply]
良いですね！ :art:

であれば、ざっくり以下のような流れになりますかね？

1. コンセプト決め：リポジトリのコンセプトやストーリー性、Contributors.jsonに入力してもらうデータを決める
2. デザインコンペ：それを元に、個別ページ、トップの表示（カード？）などのデザインをみんなでそれぞれ作ってみる
3. デザインカンプ用意：それぞれのデザインの良いところを取ったり混ぜたりして、1人がベースを整えてまとめる
4. 実装！

**Reply by pss-aileen** [reply]
↑そのような感じを考えています！

**Reply by kazzyfrog** [reply]
Discordの、今後の流れも、上記を組み合わせて置いておきます！

- バージョン１リリースまで：
  - ドッグフーディングしてみる
  - ドッグフーディングにて、見つかった修正箇所などをリストアップする（discussionの専用スレ → issue化）
  - 上記の修正＋スクショの追加
- バージョン1.0.n：リファクタリングなどの修正で、バージョン２の開発はまだ行わない
- バージョン２の仕様決め：
  1. コンセプト決め：リポジトリのコンセプトやストーリー性、Contributors.jsonに入力してもらうデータを決める
  2. デザインコンペ：それを元に、個別ページ、トップの表示（カード？）などのデザインをみんなでそれぞれ作ってみる
  3. デザインカンプ用意：それぞれのデザインの良いところを取ったり混ぜたりして、1人がベースを整えてまとめる
  4. バージョン２の実装！

**Comment by kagomen** [comment]
## トピック: デザインはスマホ版から考えるべきか？PC版から考えるべきか？
- バージョン1ではPC版から実装したが、Tailwindを使うならスマホ版から実装した方が良いのでは？と感じた

## 結論: 一言でまとめる
- デザインは先に用意する！
- 実装自体はスマホから組んでいく！

## 関連 Issues、Discussions

**Reply by pss-aileen** [reply]
同じくです！
メディアクエリーが、`min-width: 768px`、`min-width: 1024px` のように、
768pxより大きい場合、1024pxより大きい場合といった書き方なので
スマホから実装して、md:、lg: でCSSを上書きする感じが適切だと思いまいた！

**Reply by kazzyfrog** [reply]
実装、確かにそれで良いと思います！

自分はいつもは、PCとレスポンシブ同時に実装してしまうことが多いので、
バージョン２では、細かくコンポーネントを切って同時にやってもいいかもです！

> デザインはスマホ版から考えるべきか？PC版から考えるべきか？

デザインに関しては、バージョン２では、実装前にPC・スマホなどのデザインカンプを先に用意しておきたいですね！
なので、実装しながら考えることは減らせそう？です！

**Reply by pss-aileen** [reply]
> デザインに関しては、バージョン２では、実装前にPC・スマホなどのデザインカンプを先に用意しておきたいですね！
> なので、実装しながら考えることは減らせそう？です！

ですね、実装とデザインが頭でごちゃっとしてしまっていましたが

デザインは先に用意するので、その部分での迷いは少ないはず！
実装自体はスマホから組んでいく方が効率的！といったかんじですね！

**Reply by kagomen** [reply]
> デザインは先に用意
> 実装自体はスマホから組んでいく

ですね！！
これで良いと思います！👀✨

**Comment by kazzyfrog** [comment]
## 次の1週間でやること

**Reply by kazzyfrog** [reply]
Discordの、今後の流れも、上記を組み合わせて置いておきます！

- バージョン１リリースまで：
  - ドッグフーディングしてみる
  - ドッグフーディングにて、見つかった修正箇所などをリストアップする（discussionの専用スレ → issue化）
  - 上記の修正＋スクショの追加
- バージョン1.0.n：リファクタリングなどの修正で、バージョン２の開発はまだ行わない
- バージョン２の仕様決め：
  1. コンセプト決め：リポジトリのコンセプトやストーリー性、Contributors.jsonに入力してもらうデータを決める
  2. デザインコンペ：それを元に、個別ページ、トップの表示（カード？）などのデザインをみんなでそれぞれ作ってみる
  3. デザインカンプ用意：それぞれのデザインの良いところを取ったり混ぜたりして、1人がベースを整えてまとめる
  4. バージョン２の実装！

**Reply by kazzyfrog** [reply]
まず、ドッグフーディングの修正！

ディスカッションで話して、
イシュー化して、
着手！

**Reply by kazzyfrog** [reply]
バージョンのタグと、リリースノートを作る！

**Reply by kazzyfrog** [reply]
イシューは、ピン留め方式に戻す！

**Comment by pss-aileen** [comment]
@kazzyfrog @kagomen 
ちなみにリリースノートってどの媒体で出すか等決まってますか？
これから確定させていく流れでしょうか？

**Reply by kagomen** [reply]
GitHubにリリースノートを自動生成してくれる機能があるので、ひとまずこちらを使おうという方針になりました。
mdファイルなどが生成されるわけではなく、リポジトリトップページのサイドメニューにあるReleases欄が使えるようになる感じっぽいです。！
参考：
https://docs.github.com/ja/repositories/releasing-projects-on-github/automatically-generated-release-notes

**Reply by kazzyfrog** [reply]
このリポジトリの[このページ](https://github.com/first-contributions-ja/first-contributions-ja.github.io/releases)に、
バージョンタグの区切りごとに、変更内容のノートが、溜まっていく感じです！

同じように、リリースノートをGithubに載せているリポジトリの例：
https://github.com/vercel/swr/releases

**Reply by pss-aileen** [reply]
なるほど、承知しました！
